c/tskit/tables.c

```diff
@@ -372,8 +374,9 @@ write_offset_col(
         }
         type = KAS_UINT32;
         data = offset32;
+        /* We've just allocated a temp buffer, so kas can't borrow */
+        ret = kastore_puts(store, offset_col_name, data, (size_t) len, type, 0);
     }
-    ret = kastore_puts(store, offset_col_name, data, (size_t) len, type, 0);
```

Comment:
```
ret = kastore_puts(store, offset_col_name, data, (size_t) len, type, put_flags);
```

I think this is more robust and avoids the "don't check ret immediately" problem, which is always an error waiting to happen in my book.